os-eap7-launch/added/launch/elytron.sh

```diff
@@ -27,23 +27,42 @@ function configure_https() {
     return 
   fi
 
-  if [ -n "${HTTPS_PASSWORD}" -a -n "${HTTPS_KEYSTORE_DIR}" -a -n "${HTTPS_KEYSTORE}" -a -n "${HTTPS_KEYSTORE_TYPE}" ]; then
+  if [ -n "${HTTPS_PASSWORD}" -a -n "${HTTPS_KEYSTORE}" -a -n "${HTTPS_KEYSTORE_TYPE}" ]; then
 
     if [ -n "${HTTPS_KEY_PASSWORD}" ]; then
-      key_password="<credential-reference clear-text=\"${HTTPS_KEY_PASSWORD}\"/>"
+      key_password="${HTTPS_KEY_PASSWORD}"
+    else
+      echo "No HTTPS_KEY_PASSWORD was provided; using HTTPS_PASSWORD for Elytron LocalhostKeyManager."
+      key_password="${HTTPS_PASSWORD}"
+    fi
+
+    if [ -z "${HTTPS_KEYSTORE_DIR}"  ]; then
+      # Documented behavior; HTTPS_KEYSTORE is relative to the config dir
+      # Use case is the user puts their keystore in their source's 'configuration' dir and s2i pulls it in
+      keystore_path="path=\"${HTTPS_KEYSTORE}\""
+      keystore_rel_to="relative-to=\"jboss.server.config.dir\""
+    elif [[ "${HTTPS_KEYSTORE_DIR}" =~ ^/ ]]; then
+      # Assume leading '/' means the value is a FS path
+      # Standard template behavior where the template sets this var to /etc/eap-secret-volume
+      keystore_path="path=\"${HTTPS_KEYSTORE_DIR}/${HTTPS_KEYSTORE}\""
+      keystore_rel_to=""
+    else
+      # Compatibility edge case. Treat no leading '/' as meaning HTTPS_KEYSTORE_DIR is the name of a config model path
+      keystore_path="path=\"${HTTPS_KEYSTORE}\""
+      keystore_rel_to="relative-to=\"${HTTPS_KEYSTORE_DIR}\""
```

Comment:
I think it makes sense, however it doesn't correspond to the steps described in https://issues.jboss.org/browse/CLOUD-2406?focusedCommentId=13624783&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13624783 if I am not mistaken.

Reply:
@tremes Right; it's different and I should comment on the JIRA why. Thanks for reminding me!

When I wrote that JIRA comment I didn't realize that our templates were setting HTTPS_KEYSTORE_DIR to /etc/eap-secret-volume. That's not a config model logical path name and that usage shows me that the primary use case for this setting was as a FS path in https.sh.  So to be user friendly having it work as a FS path here as should be supported, with the "config model logical path name" being more of an edge case for the chance that someone got this working despite CLOUD-2406.

So, we test HTTPS_KEYSTORE_DIR for leading '/' and act accordingly.

In that comment I also talk about testing HTTPS_KEYSTORE for a leading '/' and treating it as an absolute path and not doing relative-to if it has one. But, Ken and I realized that in no existing code did we have that functionality. Adding it would basically be a minor RFE. And with the handling of HTTPS_KEYSTORE_DIR already being complex, adding yet another config permutation would have just been too confusing and bug prone. Better to keep it simple and continue to treat HTTP_KEYSTORE as a filename relative to some other path.

Reply:
I commented on the JIRA as well.